packages/meditrak-app/app/entityMenu/EntityList.js

```diff
@@ -91,50 +101,64 @@ export class EntityList extends PureComponent {
     if (!searchTerm) {
       this.setState({
         searchTerm: '',
-        searchResults: null,
+        primarySearchResults: null,
+        secondarySearchResults: null,
       });
       return;
     }
 
-    const { entities } = this.props;
-    const lowerCaseSearchTerm = searchTerm.toLowerCase();
-
-    const searchResults = entities
-      .filter(
-        ({ name, parentName }) =>
-          name.toLowerCase().includes(lowerCaseSearchTerm) ||
-          (parentName && parentName.toLowerCase().includes(lowerCaseSearchTerm)),
-      )
-      .sort(({ name: a }, { name: b }) => {
-        // Send entity names that start with the search term to the top.
-        const checkMatchesAtStart = name => name.toLowerCase().startsWith(lowerCaseSearchTerm);
-        if (checkMatchesAtStart(a) !== checkMatchesAtStart(b)) {
-          return checkMatchesAtStart(a) ? -1 : 1;
-        }
-
-        // Sort alphabetically
-        return a.localeCompare(b);
-      });
+    const primarySearchResults = this.baseQuery.filtered(`name BEGINSWITH[c] $0`, searchTerm);
+    const secondarySearchResults = this.baseQuery.filtered(
+      `(NOT name BEGINSWITH[c] $0) AND (name CONTAINS[c] $0 OR parent.name BEGINSWITH[c] $0)`,
+      searchTerm,
+    );
     this.setState({
       searchTerm,
-      searchResults,
+      primarySearchResults,
+      secondarySearchResults,
     });
   };
 
-  getListSections = () => {
-    const { searchResults } = this.state;
-    if (searchResults) {
-      return [{ data: searchResults }];
+  getListData = () => {
+    const { primarySearchResults, secondarySearchResults, numberToShow } = this.state;
+
+    if (primarySearchResults || secondarySearchResults) {
+      const totalResultsAvailable = primarySearchResults.length + secondarySearchResults.length;
+      if (totalResultsAvailable === 0) {
+        return { noResults: true };
+      }
+      const data = [
+        ...primarySearchResults.slice(0, numberToShow),
+        ...secondarySearchResults.slice(0, numberToShow - primarySearchResults.length),
+      ];
```

Comment:
Hmmm won't this go a little wrong if `primarySearchResults.length > numberToShow`?
```suggestion
      const data = [
        ...primarySearchResults.slice(0, numberToShow),
        ...secondarySearchResults.slice(0, Math.max(0, numberToShow - primarySearchResults.length)),
      ];
```

Reply:
Great catch!!